### PR TITLE
Configuration details

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

> Heads up 👷: this PR will be split into multiple for easier review-ability. Stay calm while reading the size/XXL label.

Renames Repositories to Configurations and adds basic settings, migrated from the Projects Settings page.

<details>
<summary>Summary generated by Copilot</summary>

</details>

## Related Issue(s)

Fixes EXP-800
Fixes EXP-801

## How to test
Visit the Configurations page, which is found under the org dropdown on the top left.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-configu16257aa40a</li>
	<li><b>🔗 URL</b> - <a href="https://ft-configu16257aa40a.preview.gitpod-dev.com/workspaces" target="_blank">ft-configu16257aa40a.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-configurations-details-gha.18822</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-configu16257aa40a%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

### CODE REVIEW

File: components/dashboard/src/configurations/detail/ConfigurationVariables.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+                {action === "add" && (
+                    <Alert type="warning">
+                        <strong>Configuration environment variables can be exposed.</strong>
+                        <br />
+                        Even if <strong>Hide Variable in Workspaces</strong> is enabled, anyone with read access to your
+                        repository can access secret values if they are printed in the terminal, logged, or persisted to
+                        the file system.
+                    </Alert>
+                )}
+                <div className="mt-8">
```
Comment (selfcontained) [inline components/dashboard/src/configurations/detail/ConfigurationVariables.tsx:168]: let's use the `TextInputField` component here.

File: components/dashboard/src/configurations/detail/ConfigurationVariables.tsx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                        autoFocus={action === "add"}
+                        className="w-full"
+                        type="text"
+                        name="name"
+                        value={name}
+                        readOnly={action === "update"}
+                        onChange={(e) => setName(e.target.value)}
+                    />
+                </div>
+                <div className="mt-4">
```
Comment (selfcontained) [inline components/dashboard/src/configurations/detail/ConfigurationVariables.tsx:180]: `TextInputField` here

File: components/dashboard/src/configurations/detail/ConfigurationWorkspaces.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                    },
+                },
+            );
+        },
+        [repository, toast, updateRepository],
+    );
+
+    const setWorkspaceClass = useCallback(
+        async (value: string) => {
+            if (!repository) {
```
Comment (selfcontained) [inline components/dashboard/src/configurations/detail/ConfigurationWorkspaces.tsx:44]: not needed

File: components/dashboard/src/configurations/detail/ConfigurationWorkspaces.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    repository: Project;
+}
+
+export default function RepositoryWorkspacesSettings({ repository }: RepositoryPrebuildsSettingsProps) {
+    const updateRepository = useUpdateProject();
+    const { toast } = useToast();
+
+    const updateRepositorySettings = useCallback(
+        async (settings: ProjectSettings) => {
+            if (!repository) return;
```
Comment (selfcontained) [inline components/dashboard/src/configurations/detail/ConfigurationWorkspaces.tsx:24]: not needed

File: components/dashboard/src/configurations/detail/ConfigurationWorkspaces.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    const setWorkspaceClass = useCallback(
+        async (value: string) => {
+            if (!repository) {
+                return value;
+            }
+            const before = repository.settings?.workspaceClasses?.regular;
+            updateRepositorySettings({
+                workspaceClasses: { ...repository.settings?.workspaceClasses, regular: value },
+            });
+            return before;
```
Comment (selfcontained) [inline components/dashboard/src/configurations/detail/ConfigurationWorkspaces.tsx:51]: This return value doesn't seem to be used for anything, lets remove if we can.

**Review by selfcontained — CHANGES_REQUESTED** [review]
Left a bunch of comments. The biggest concern I have atm with these changes is around the prebuild settings ux - it's just really awkward that we're changing all the values on the fly. Lots of toasts, occasional values in inputs changing as you type due to api calls happening as you type. I think this particular section could really benefit from working with Cory on the design/UX for how we update/change these values. There's a lot of complexity in that part of the UI that comes from how it currently behaves that I think we could simplify if we approached it a little different.

File: components/dashboard/src/components/podkit/drop-down/DropDown.tsx
Comment (selfcontained) [inline components/dashboard/src/components/podkit/drop-down/DropDown.tsx:1]: Seems like the background color isn't getting applied to the dropdown content. You can see the hamburger icon underneath it like in this screenshot.
image

File: components/dashboard/src/configurations/detail/ConfigurationDeleteModal.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+export const RemoveConfigurationModal: FunctionComponent<RemoveConfigurationModalProps> = ({
+    configuration,
+    onClose,
+    onRemoved,
+}) => {
+    const [disabled, setDisabled] = useState(false);
+    const useDeleteRepository = useDeleteProject();
+
+    const removeConfiguration = useCallback(async () => {
+        setDisabled(true);
```
Comment (selfcontained) [inline components/dashboard/src/configurations/detail/ConfigurationDeleteModal.tsx:27]: We shouldn't need to track this state, we can use the query `isLoading` prop for this instead.

File: components/dashboard/src/menu/OrganizationSelector.tsx [hunk 7]
```diff
@@ -60,11 +60,11 @@ export default function OrganizationSelector() {
     if (currentOrg.data) {
         if (repoConfigListAndDetail) {
             linkEntries.push({
-                title: "Repositories",
-                customContent: <LinkEntry>Repositories</LinkEntry>,
+                title: "Configurations",
```
Comment (selfcontained) [inline components/dashboard/src/menu/OrganizationSelector.tsx:63]: I didn't think we wanted to call this Configurations in the UI and we're leaving it as Repositories. Same w/ the actual URL - `/repositories`

File: components/dashboard/src/configurations/detail/ConfigurationDelete.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    configuration: Project;
+}
+
+export default function DeleteConfiguration({ configuration }: RepositoryPrebuildsSettingsProps) {
+    const [showRemoveModal, setShowRemoveModal] = useState(false);
+
+    const history = useHistory();
+
+    const onProjectRemoved = useCallback(() => {
+        history.push("/repositories");
```
Comment (selfcontained) [inline components/dashboard/src/configurations/detail/ConfigurationDelete.tsx:23]: This url doesn't work with it being changed to `/configuration` - but I think we should leave it as `/repositories` anyways.

File: components/dashboard/src/configurations/detail/ConfigurationDetail.tsx [hunk 9]
```diff
@@ -42,11 +46,17 @@ const RepositoryDetailPage: FC = () => {
                         // TODO: add a better not-found UI w/ link back to repositories
                         <div>Sorry, we couldn't find that repository configuration.</div>
                     ) : (
-                        <RepositoryNameForm project={data} />
+                        <>
+                            <ConfigurationNameForm configuration={data} />
+                            <ConfigurationEnvironmentVariables configuration={data} />
+                            <RepositoryPrebuildsSettings repository={data} />
```
[the remark this answers is not in the document]
Comment (filiptronicek) [reply components/dashboard/src/configurations/detail/ConfigurationDetail.tsx:52]: Of course, thanks for catching this!

**Comment by filiptronicek** [comment]
Continuing in https://github.com/gitpod-io/gitpod/pull/19021

**Pull request closed without merging**